File: src/main/java/todo/entity/Task.java
package todo.entity;

import java.util.Date;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import lombok.Data;

@Data
@Entity
@Table(name = "tasks")
public class Task {
    
    @Id
    @GeneratedValue
    private int id;

    private int folderId;

    private String title;

    private int status;

    @Temporal(TemporalType.DATE)
    private Date dueDate;

    @Temporal(TemporalType.TIMESTAMP)
    private Date createdAt;

    @Temporal(TemporalType.TIMESTAMP)
    private Date updatedAt;

    @ManyToOne
    private Folder folder;

}

File: model/task.go
package model

import "time"

type Task struct {
	id          uint         `json:"id" gorm:"primaryKey"`
	folderId    uint         `json:"folderId"`
	title       string       `json:"title"`
	status      int          `json:"status"`
	dueDate     time.Time    `json:"dueDate"`
	createdAt   time.Time    `json:"createdAt"`
	updatedAt   time.Time    `json:"updatedAt"`
}